server/src/main/java/com/objectcomputing/checkins/services/file/FileServicesImpl.java

```diff
@@ -0,0 +1,226 @@
+
+        try {
+            Set<FileInfoDTO> result = new HashSet<>();
+            Drive drive = googleDriveAccessor.accessGoogleDrive();
+            validate(drive == null, "Unable to access Google Drive");
+
+            if (checkInID == null && isAdmin) {
+                //find all
+                FileList fileList = drive.files().list().execute();
+                for (File file : fileList.getFiles()) {
+                    result.add(setFileInfo(file, null));
+                }
+            } else if (checkInID != null) {
+                //find by CheckIn ID
+                CheckIn checkIn = checkInServices.read(checkInID);
+                validate(checkIn == null, "Unable to find checkin record with id %s", checkInID);
+
+                Set<CheckinDocument> checkinDocuments = checkinDocumentServices.read(checkInID);
+                for (CheckinDocument cd : checkinDocuments) {
+                    File file = drive.files().get(cd.getUploadDocId()).execute();
+                    result.add(setFileInfo(file, cd));
+                }
+            }
+
+            return HttpResponse.ok(result);
+        } catch (GoogleJsonResponseException e) {
+            LOG.error("Error occurred while retrieving files from Google Drive.", e);
+            return HttpResponse
+                    .status(HttpStatus.valueOf(e.getStatusCode()))
+                    .body(e.getContent());
+        } catch (IOException e) {
+            LOG.error("Error occurred while retrieving files from Google Drive.", e);
+            return HttpResponse.status(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @Override
+    public HttpResponse<?> downloadFiles(@NotNull String uploadDocId) {
+
+        try {
+            OutputStream outputStream = new ByteArrayOutputStream();
+            java.io.File file = java.io.File.createTempFile("tmp", ".txt");
+            file.deleteOnExit();
+            FileWriter myWriter = new FileWriter(file);
+
+            Drive drive = googleDriveAccessor.accessGoogleDrive();
+            validate(drive == null, "Unable to access Google Drive");
+
+            drive.files().get(uploadDocId).executeMediaAndDownloadTo(outputStream);
+            myWriter.write(String.valueOf(outputStream));
+            myWriter.close();
+
+            return HttpResponse.status(HttpStatus.OK).body(file);
+        }  catch (HttpResponseException e) {
+            LOG.error("Error occurred while retrieving files from Google Drive.", e);
+            return HttpResponse
+                    .status(HttpStatus.valueOf(e.getStatusCode()))
+                    .body(e.getContent());
+        } catch (IOException e) {
+            LOG.error("Error occurred while retrieving files from Google Drive.", e);
+            return HttpResponse.status(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @Override
+    public HttpResponse<FileInfoDTO> uploadFile(@NotNull UUID checkInID, @NotNull CompletedFileUpload file) {
```

Comment:
Also missing from the DoD...upload should only be possible by the same folk as above (current PDL, checkin PDL, or ADMIN). sorry. :-( Those missing items can be made into a new issue if you'd prefer. Just please do so before merging/closing this one.

Reply:
Not a problem, it shouldn't be a big change.

Reply:
@mkimberlin - Should we block user from uploading a new document if checkin is completed?

Reply:
updated in the latest commit.